src/components/IconToggle/IconToggle.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import {
+  Checkbox as CoreCheckbox,
+  CheckboxProps as CoreCheckboxProps
+} from "wix-ui-core/checkbox";
+import styles from "./IconToggle.st.css";
+import { TPAComponentsConsumer } from "../TPAComponentsConfig";
+
+import { ReactComponent as HeartIcon } from "../../icons/raw/Heart.svg";
+
+export interface OnChangeEvent extends React.ChangeEvent<HTMLInputElement> {
+  checked: boolean;
+}
+
+export type Placement = "right" | "left";
+
+export interface IconToggleProps {
+  icon: React.ReactNode;
+  label: React.ReactNode | string;
+  labelPlacement: Placement;
+  onChange?: (event: OnChangeEvent) => void;
+  checked: boolean;
+  disabled: boolean;
+}
+
+interface DefaultProps {
+  icon: React.ReactNode;
+  label: React.ReactNode | string;
+  labelPlacement: Placement;
+  checked: boolean;
+  disabled: boolean;
+}
+
+/** IconToggle */
+export class IconToggle extends React.Component<IconToggleProps> {
+  static displayName = "IconToggle";
+  static defaultProps: DefaultProps = {
+    icon: <HeartIcon />,
+    label: "Like",
+    labelPlacement: "right",
+    checked: false,
+    disabled: false
+  };
+
+  render() {
+    const {
+      icon,
+      label,
+      labelPlacement,
+      onChange,
+      checked,
+      disabled,
+      ...rest
+    } = this.props;
+
+    const renderIcon = () => (
+      <div className={styles.container}>
+        <div className={styles.icon}>{icon}</div>
+        <div className={styles.label}>{label}</div>
+      </div>
+    );
+
+    return (
+      <TPAComponentsConsumer>
+        {({ mobile }) => (
```

Comment:
The generator generates the components with this `mobile` context prop.
This is used for some components, that have a different design in mobile sites.
Since we don't have a special design or behaviour on mobile, I think we can remove all the mobile references, because it's just adding noise to the code.
Sorry for the mess, we might want to remove the `mobile` prop from the generator, in the future, so developers won't have to go back and erase it wherever it's not needed.

Reply:
mobile prop and style removed.